src/octopeer-github/main/MainController.ts

```diff
@@ -118,27 +113,62 @@ class MainController implements OptionsObserver {
         return eventObject;
     }
 
+    private postToDatabase(message: EventObject) {
+        const success = function() {
+            Logger.debug(`Successfully logged to database: ${JSON.stringify(message)}`);
+        };
+        const failure = function() {
+            Logger.warn("Log to database of following object failed:");
+            Logger.warn(message);
+        };
+        this.database.post(message, success, failure);
+    }
+
     /**
      * Only sends a message to a tab if its URL belongs to a Pull Request.
      * I named it alike to a "test-and-set" operation that comes from concurrent programming.
      *     This (atomic) operation only sets a variable if a condition holds.
-     * @param tab   the Tab to check for
+     * @param tab           the Tab to check for
+     * @param isActiveTab   if `tab` is not the active tab, this method will not update the Status.
      */
-    private testAndSend(tab: Tab) {
+    private testAndSend(tab: Tab, isActiveTab = true) {
         if (tab === undefined || tab.url === undefined) {
-            Status.standby();
+            this.setNewStatus(false, isActiveTab);
             return;
         }
-        const url = tab.url;
-        let urlInfo = URLHandler.isPullRequestUrl(url);
-        if (urlInfo.equals([])) {
-            Status.standby();
+
+        let isPullRequest = URLHandler.isPullRequestUrl(tab.url);
+        if (isPullRequest) {
+            this.database = new RESTApiDatabaseAdapter("http://146.185.128.124", tab.url, "Travis");
+            this.sendMessageToContentScript(tab, Options.get(Options.LOGGING) && isPullRequest);
+        }
+        this.setNewStatus(isPullRequest, isActiveTab);
+    }
+
+    /**
+     * Sets the status to STANDBY or RUNNING based on isPullRequest.
+     * If isActiveTab or Options.LOGGING is false, nothing will happen.
+     * If new status is not equal to the previous status, post a start-/stop-watching PR event to the database.
+     * @param isPullRequest whether this method should behave like the handled tab is a PR.
+     * @param isActiveTab   whether this method should behave like the handled tab is active.
+     */
+    private setNewStatus(isPullRequest: boolean, isActiveTab: boolean) {
+        if (!isActiveTab || !Options.get(Options.LOGGING)) {
             return;
         }
-        Status.running();
-        Logger.debug(`[Tab] Owner: ${urlInfo[1]}, Repo: ${urlInfo[2]}, PR-number: ${urlInfo[3]}`);
+        let eventID = isPullRequest ? EventID.START_WATCHING_PR : EventID.STOP_WATCHING_PR;
+        let status =  isPullRequest ? StatusCode.RUNNING        : StatusCode.STANDBY;
+        let wasItRunning = Status.isStatus(StatusCode.RUNNING);
```

Comment:
These values can indeed be `const`. But, the following if-statement is easier to understand if the naming is this way. Because the status is set at the end of this function, here the reasoning is done with the "old status", and it is checked whether is "was running" before the status change.